server/middlewares/validations.js

```diff
@@ -0,0 +1,77 @@
+const validateUsername = (username) => {
+  const usernamePattern = /^[a-zA-Z]+[a-zA-Z0-9_]+$/;
+  return usernamePattern.test(username);
+};
+
+const validateEmail = (email) => {
+  const emailPattern = /^[a-zA-Z0-9]+[-_.]*[a-zA-Z0-9]*@[a-zA-Z0-9]+\.(?:[a-zA-Z0-9]+)*$/;
+  return emailPattern.test(email);
+};
+
+const validatePassword = (password) => {
+  const passwordPattern = /^[0-9a-z]+$/;
+  return passwordPattern.test(password);
+};
+
+const checkMissingFields = (requestObject, requiredFieldObject) => {
+  const output = {
+    isValid: true,
+    problem: {},
+  };
+  for (let i = 0; i < requiredFieldObject.length; i += 1) {
+    if (requestObject[requiredFieldObject[i]] === undefined) {
+      if (output.isValid === true) {
+        output.problem.invalidCredentials = '';
+      }
+      output
+        .problem
+        .invalidCredentials += `The ${requiredFieldObject[i]} is empty. `;
+      output.isValid = false;
+    }
+  }
+  if (output.isValid === false) {
+    return output;
+  }
+  if (requestObject.password.length < 8) {
+    output.isValid = false;
+    output
+      .problem
+      .passwordLength = 'The password length should be atleast 8 characters. ';
+  }
+  if (!validateUsername(requestObject.username)) {
+    output.isValid = false;
+    output
+      .problem
+      .username = 'The username must begin with an alphabet and may contain a number but not symbols. ';
+  }
+  if (!validateEmail(requestObject.email)) {
+    output.isValid = false;
+    output
+      .problem
+      .email = 'The Email provided is invalid. ';
+  }
+  if (!validatePassword(requestObject.password)) {
+    output.isValid = false;
+    output
+      .problem
+      .password = 'Password must not contain symbols. ';
+  }
+  return output;
+};
+
+const validateRequest = (req, res, next) => {
+  const { isValid, problem } = checkMissingFields(req.body, [
```

Comment:
We should be validating all the fields in the request body, pretty sure it's more than `username`, `email` and `password`.

Reply:
Based on the interaction we just had, is this still a concern? I say this because upon registration this information is all we require from the user.

Reply:
It's no longer a concern

Reply:
Thanks